lean4-infoview/src/infoview/goals.tsx

```diff
@@ -1,8 +1,28 @@
 import * as React from 'react'
 import { DocumentPosition } from './util'
-import { ConfigContext } from './contexts'
 import { InteractiveCode } from './interactiveCode'
-import { InteractiveGoal, InteractiveGoals, InteractiveHypothesis, TaggedText_stripTags } from './rpcInterface'
+import { InteractiveGoal, InteractiveGoals, InteractiveHypothesisBundle, InteractiveHypothesisBundle_accessableNames, TaggedText_stripTags } from './rpcInterface'
+
+interface HypProps {
+    pos: DocumentPosition
+    hyp: InteractiveHypothesisBundle
+    index: number
+}
+
+export function Hyp({ pos, hyp : h, index }: HypProps) {
+    const names = InteractiveHypothesisBundle_accessableNames(h).map((n, i) =>
+            <span className="mr1">{n}</span>
+        )
+    const hypKey = (h.fvarIds?.[0] ?? index)
```

Comment:
```suggestion
    const hypKey = index
```

I'm not sure if it's a good idea to include "random" information in a key.  But this is should not be *too* bad, since the hypothesis fvarids remain somewhat stable if you move along a proof.

Reply:
removed this for now as key only matters on parent `Hyp` component.